job-status-service/src/main/java/au/org/aodn/aws/wps/lambda/JobStatusServiceRequestHandler.java

```diff
@@ -0,0 +1,368 @@
+package au.org.aodn.aws.wps.lambda;
+
+import au.org.aodn.aws.util.JobFileUtil;
+import au.org.aodn.aws.util.S3Utils;
+import au.org.aodn.aws.wps.JobStatusFormatEnum;
+import au.org.aodn.aws.wps.JobStatusRequest;
+import au.org.aodn.aws.wps.JobStatusRequestParameterParser;
+import au.org.aodn.aws.wps.JobStatusResponse;
+import au.org.aodn.aws.wps.status.QueuePosition;
+import au.org.aodn.aws.wps.status.WpsConfig;
+import com.amazonaws.services.batch.AWSBatch;
+import com.amazonaws.services.batch.AWSBatchClientBuilder;
+import com.amazonaws.services.batch.model.DescribeJobsRequest;
+import com.amazonaws.services.batch.model.DescribeJobsResult;
+import com.amazonaws.services.batch.model.JobStatus;
+import com.amazonaws.services.batch.model.JobSummary;
+import com.amazonaws.services.batch.model.ListJobsRequest;
+import com.amazonaws.services.batch.model.ListJobsResult;
+import com.amazonaws.services.lambda.runtime.Context;
+import com.amazonaws.services.lambda.runtime.RequestHandler;
+import com.amazonaws.services.s3.AmazonS3;
+import com.amazonaws.services.s3.AmazonS3ClientBuilder;
+import com.amazonaws.util.StringInputStream;
+import net.opengis.wps._1_0.ExecuteResponse;
+import net.opengis.wps._1_0.ProcessFailedType;
+import net.opengis.wps._1_0.ProcessStartedType;
+import net.opengis.wps._1_0.StatusType;
+import org.apache.http.HttpStatus;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.xml.bind.JAXBContext;
+import javax.xml.bind.JAXBException;
+import javax.xml.bind.util.JAXBSource;
+import javax.xml.transform.Transformer;
+import javax.xml.transform.TransformerException;
+import javax.xml.transform.TransformerFactory;
+import javax.xml.transform.stream.StreamResult;
+import javax.xml.transform.stream.StreamSource;
+import java.io.IOException;
+import java.io.StringWriter;
+import java.io.UnsupportedEncodingException;
+import java.util.ArrayList;
+
+public class JobStatusServiceRequestHandler implements RequestHandler<JobStatusRequest, JobStatusResponse> {
+
+    private static final JobStatusFormatEnum DEFAULT_FORMAT = JobStatusFormatEnum.XML;
+
+    private Logger LOGGER = LoggerFactory.getLogger(JobStatusServiceRequestHandler.class);
+
+    @Override
+    public JobStatusResponse handleRequest(JobStatusRequest request, Context context) {
+
+        JobStatusResponse response;
+        JobStatusResponse.ResponseBuilder responseBuilder = new JobStatusResponse.ResponseBuilder();
+        String responseBody;
+
+        JobStatusRequestParameterParser parameterParser = new JobStatusRequestParameterParser(request);
+        String jobId = parameterParser.getJobId();
+        String format = parameterParser.getFormat();
+
+        LOGGER.info("Parameters passed: JOBID [" + jobId + "], FORMAT [" + format + "]");
```

Comment:
Format could be determined by requested ContentType?

Reply:
Other OGC services use format as well so can't see a problem with doing it here as well.  Its not really using REST style API.